### PR TITLE
bump oras.land/oras-go to v1.2.6

### PR DESCRIPTION
**Pull request by tariq1890**
**What this PR does / why we need it**:

This PR bumps the github.com/docker/docker dependency imported transitively through the `oras-go` dependency.

This PR should address the following critical CVE:

https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-41110

GHSA link: https://github.com/advisories/GHSA-v23v-6jw2-98fq

### CODE REVIEW

File: pkg/registry/client_http_test.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"os"
 	"testing"
 
-	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/errdefs"
```
Comment (tariq1890) [inline pkg/registry/client_http_test.go:24]: This change is needed to fix this error

```
Error: pkg/registry/client_http_test.go:24:2: SA1019: "github.com/containerd/containerd/errdefs" is deprecated: use [github.com/containerd/errdefs]. (staticcheck)
  	"github.com/containerd/containerd/errdefs"
  ```

**Comment by tariq1890** [comment]
Hi @mattfarina can we merge this?

**Comment by scottrigby** [comment]
Hi Tariq. The CVE was a false positive, but addressed by a smaller version bump here: https://github.com/helm/helm/pull/13226

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module helm.sh/helm/v3
 
-go 1.22.0
+go 1.22.4
```
Comment (scottrigby) [inline go.mod:3]: We generally only bump the Go version as required for newer kubernetes libs, for required security fixes. Check git history for examples. This message from a previous upgrade explains why:

> The version specified in the go.mod file is the minimum version we expect Helm
to be compiled against. This is the oldest supported version to support environments where others compile Helm.

**Pull request closed without merging**